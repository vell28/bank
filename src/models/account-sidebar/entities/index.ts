export interface IAccountSidebar {
  [key: string]: boolean;
}
