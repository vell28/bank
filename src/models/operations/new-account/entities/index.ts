export enum AccountTypes {
  LINKED = 'LINKED',
  BUSINESS = 'BUSINESS',
  SAVINGS = 'SAVINGS',
}

export type AccountTypeType =
  | AccountTypes.LINKED
  | AccountTypes.BUSINESS
  | AccountTypes.SAVINGS;

export enum AccountNames {
  LINKED = 'Linked',
  BUSINESS = 'Business',
  SAVINGS = 'Savings',
}

export type AccountNameType =
  | AccountNames.LINKED
  | AccountNames.BUSINESS
  | AccountNames.SAVINGS;

export interface IAccount {
  type: AccountTypeType;
  name: AccountNameType;
}

export const ACCOUNTS: IAccount[] = [
  {
    type: AccountTypes.LINKED,
    name: AccountNames.LINKED,
  },
  {
    type: AccountTypes.BUSINESS,
    name: AccountNames.BUSINESS,
  },
  {
    type: AccountTypes.SAVINGS,
    name: AccountNames.SAVINGS,
  },
];
