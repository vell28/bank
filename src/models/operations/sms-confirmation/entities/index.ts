export interface IConfirmation {
  confirmationCodeLength?: number;
  confirmationToken?: string;
  attempts?: number;
}
