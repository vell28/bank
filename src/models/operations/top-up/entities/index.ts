export interface ICardData {
  cardNumber?: string;
  beneficiaryName?: string;
  expiresDate?: string;
  isHolder?: boolean;
  cvv?: string;
  country?: string;
  city?: string;
}

export interface ITopUpRequest {
  email?: string;
  initiatorName?: string;
}

export interface ITopUp {
  card: ICardData;
  value: string;
  redirectUrl?: string;
  description?: string;
  request: ITopUpRequest;
}
