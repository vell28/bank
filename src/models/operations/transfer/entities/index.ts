export interface ICardData {
  cardNumber?: string;
  beneficiaryName?: string;
  expiresDate?: string;
  cvv?: string;
}

export interface IBankData {
  beneficiaryName?: string;
  iban?: string;
  country?: string;
  city?: string;
  address?: string;
  swiftCode?: string;
  beneficiaryBank?: string;
  isIntermediaryBank?: boolean;
  intermediarySwift?: string;
  intermediaryBank?: string;
  intermediaryAccount?: string;
  commissionType?: SwiftCommissionType;
  urgency?: SwiftUrgencyType;
  purpose?: string;
  transferDetails?: string;
}

export interface IRate {
  currencyFrom: string;
  currencyTo: string;
  fromAmount2: number;
  fromAmount3:number;
  fromAmount4: number;
  rate: number;
  rate2: number;
  rate3: number;
  rate4: number;
}

export type IRates = IRate | [];

export interface ITransfer {
  card: ICardData;
  phone: string;
  value: string;
  purpose: string;
  bank: IBankData;
  rates: IRates;
}

export enum SwiftCommissionTypes {
  BENEFICIARY = 'BENEFICIARY',
  SHARED = 'SHARED',
  SENDER = 'SENDER',
}

export type SwiftCommissionType =
  | SwiftCommissionTypes.BENEFICIARY
  | SwiftCommissionTypes.SHARED
  | SwiftCommissionTypes.SENDER;

export enum SwiftUrgencyTypes {
  NORMAL = 'Normal',
  URGENT = 'Urgent',
}

export type SwiftUrgencyType = SwiftUrgencyTypes.NORMAL | SwiftUrgencyTypes.URGENT;
