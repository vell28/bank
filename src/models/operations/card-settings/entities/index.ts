import { CurrencyCodeType } from 'modules/currencies';
import { AccountStatusType, IBalance } from 'models/organizations/entities';

export enum CardProduct {
  MAIN = 'MAIN',
  ADDITIONAL = 'ADDITIONAL',
  FAMILY = 'FAMILY',
  KIDS = 'KIDS',
}

export type CardProductType = CardProduct.MAIN | CardProduct.ADDITIONAL | CardProduct.FAMILY | CardProduct.KIDS;

export interface ICardOwner {
  embossedName: string;
}

export enum Limit {
  DAY = 'DAY',
  MONTH = 'MONTH',
  FREE_ATM_WITHDRAWAL = 'FREE_ATM_WITHDRAWAL',
}

export type LimitType = Limit.DAY | Limit.MONTH | Limit.FREE_ATM_WITHDRAWAL;

export interface ILimit {
  type: LimitType;
  title: null | string;
  maxValue: number;
  usedValue: number;
  preferredMaxValues?: number[];
}

export interface ILimitData {
  type: LimitType;
  maxValue: string;
}

export interface ICardOptions {
  limits?: {
    disable?: boolean;
    disable5min?: boolean;
    disableOneTime?: boolean;
  };
}

export interface ICard {
  accountId: string;
  availableBalance: number;
  backgroundImage?: string;
  balance: number;
  balanceHold: number;
  balances: IBalance[];
  clientId: string;
  createdAt?: string;
  currency: {
    code: CurrencyCodeType;
  };
  expireAt?: string;
  id: string;
  limits: ILimit[];
  number?: string;
  options: ICardOptions;
  owner: ICardOwner;
  paySystem: string;
  plasticStatus: string;
  productType: CardProductType;
  status: AccountStatusType;
  title?: string;
}

export interface ICardUnmasked {
  number: string;
  pin: string | null;
  cvc: string;
  expireAt: string;
  owner: ICardOwner;
}
