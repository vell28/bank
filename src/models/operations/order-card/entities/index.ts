import { CardProductType, ICardOwner } from '../../card-settings/entities/index';

export interface ICardOrderOwner extends ICardOwner {
  phoneNumber: string;
}

export interface IAddress {
  apartmentNumber: number;
  city: string;
  countryCode: string;
  postalCode: string;
  province?: string;
  street: string;
  streetNumber: string;
}

export interface ICardOrder {
  cardProductType?: CardProductType;
  cardOwner?: ICardOrderOwner;
  deliveryAddress?: IAddress;
}

export enum CardFormat {
  VIRTUAL = 'VIRTUAL',
  PLASTIC = 'PLASTIC',
}

export type CardFormatType = CardFormat.VIRTUAL | CardFormat.PLASTIC;

export interface IOrderRequestBody {
  type: CardProductType;
  format: CardFormatType;
  cardHolderName: string;
  cardHolderPhoneNumber: string;
  isExpressDelivery: boolean;
  deliveryAddress: IAddress;
}
