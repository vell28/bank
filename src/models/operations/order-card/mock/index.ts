import { IServerAddress } from 'models/organizations/redux/actions/entities';
import { IClient } from 'models/organizations/entities';
import { IAddress, ICardOrderOwner } from '../entities';

export const exampleAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleOrganizationAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleClientInfo: ICardOrderOwner = {
  embossedName: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};

export const exampleServerAddress: IServerAddress = {
  country: 'Germany',
  province: 'North Rhine-Westphalia',
  city: 'Düsseldorf',
  postalCode: '40479',
  street: 'Gartenstraße',
  streetNumber: '53',
  apartmentNumber: '42',
};

export const exampleClient: IClient = {
  address: exampleOrganizationAddress,
  name: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};
