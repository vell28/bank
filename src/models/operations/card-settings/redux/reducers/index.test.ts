import { cardData } from 'models/organizations/mock';
import { cardSettings, ICardSettingsState } from '.';
import { lockedStatus } from '../../../../organizations/entities';
import {
  IFetchUnmaskedAction,
  FETCH_CARD_UNMASKED,
  SET_SELECTED_CARD,
  IToggleCardModal,
  LOCK_CARD,
  UNLOCK_CARD_CONFIRM,
  TOGGLE_SHOWN_UNBLOCK_CONFIRM_CARD,
  TOGGLE_SHOWN_UNBLOCK_CARD,
  TOGGLE_SHOWN_BLOCK_CARD,
  TOGGLE_SHOWN_DISABLE_CARD_LIMITS,
  TOGGLE_SHOWN_DISABLE_CARD_LIMITS_CONFIRM,
  TOGGLE_SHOWN_INCREASE_CARD_LIMIT,
  TOGGLE_SHOWN_INCREASE_CARD_LIMIT_CONFIRM,
  TOGGLE_SHOWN_CARD_LIMITS_TIMER,
  ILockCardAction,
  IIncreaseCardLimitAction,
  ISetIncreasedLimitsAction,
  DISABLE_CARD_LIMITS_CONFIRM,
  INCREASE_CARD_LIMIT_CONFIRM,
  REDUCTION_CARD_LIMIT,
  SET_INCREASED_CARD_LIMIT,
} from '../actions';

import { ICard, Limit } from '../../entities';

const initCardState: ICardSettingsState = {
  isLoading: false,
  isSuccess: false,
  isError: false,
  data: {},
  selectedCard: {},
  modals: {
    isShownLock: true,
    isShownUnLock: true,
    isShownUnLockConfirm: true,
    isShownDisableLimits: true,
    isShownDisableLimitsConfirm: true,
    isShownIncreaseLimit: true,
    isShownIncreaseLimitConfirm: true,
  },
  errors: [],
  limitsSettings: {
    isShownDisableLimitsTimer: true,
    increasedLimit: { type: Limit.DAY, maxValue: '' },
  },
};

const unmaskedData = {
  number: '**** **10 0004 ****',
  pin: null,
  cvc: '675',
  expireAt: '2022-05-01T00:00:00',
  owner: {
    embossedName: 'RALF WILLIAMS',
  },
};

const cardLimit = {
  type: Limit.DAY,
  maxValue: '500',
};

const cardId = 'test123';
const accountId = 'account123';

describe('cardSettings reducers', () => {
  test('cardSettings/FETCH_CARD_UNMASKED', () => {
    const action: IFetchUnmaskedAction = {
      type: `${FETCH_CARD_UNMASKED}_SUCCESS`,
      payload: { data: unmaskedData },
      meta: cardId,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.data[cardId]).toBe(unmaskedData);
    expect(newState).toMatchSnapshot();
  });
  test('cardSettings/SET_SELECTED_CARD', () => {
    const action = {
      type: SET_SELECTED_CARD,
      payload: {
        [accountId]: {
          index: 1,
          card: cardData,
        },
      },
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.selectedCard[accountId].index).toEqual(1);
    expect(newState.selectedCard[accountId].card).toBe(cardData);
  });
  test('cardSettings/TOGGLE_SHOWN_UNBLOCK_CONFIRM_CARD', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_UNBLOCK_CONFIRM_CARD,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownUnLockConfirm).toBeFalsy();
  });
  test('cardSettings/TOGGLE_SHOWN_BLOCK_CARD', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_BLOCK_CARD,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownLock).toBeFalsy();
  });
  test('cardSettings/TOGGLE_SHOWN_UNBLOCK_CARD', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_UNBLOCK_CARD,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownUnLock).toBeFalsy();
  });
  test('cardSettings/LOCK_CARD', () => {
    const blockedCard: ICard = {
      ...cardData,
      status: lockedStatus,
    };
    const action: ILockCardAction = {
      type: `${LOCK_CARD}_SUCCESS`,
      payload: { data: blockedCard },
      meta: accountId,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.selectedCard[accountId].card).toBe(blockedCard);
    expect(newState).toMatchSnapshot();
  });

  test('cardSettings/UNLOCK_CARD_CONFIRM', () => {
    const action: ILockCardAction = {
      type: `${UNLOCK_CARD_CONFIRM}_SUCCESS`,
      payload: { data: cardData },
      meta: accountId,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.selectedCard[accountId].card).toBe(cardData);
    expect(newState.modals.isShownUnLockConfirm).toBeFalsy();
    expect(newState).toMatchSnapshot();
  });

  test('cardSettings/DISABLE_CARD_LIMITS_CONFIRM', () => {
    const action: ILockCardAction = {
      type: `${DISABLE_CARD_LIMITS_CONFIRM}_SUCCESS`,
      payload: { data: cardData },
      meta: accountId,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.selectedCard[accountId].card).toBe(cardData);
    expect(newState.modals.isShownDisableLimitsConfirm).toBeFalsy();
    expect(newState).toMatchSnapshot();
  });

  test('cardSettings/INCREASE_CARD_LIMIT_CONFIRM', () => {
    const action: ILockCardAction = {
      type: `${INCREASE_CARD_LIMIT_CONFIRM}_SUCCESS`,
      payload: { data: cardData },
      meta: accountId,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.selectedCard[accountId].card).toBe(cardData);
    expect(newState.modals.isShownIncreaseLimitConfirm).toBeFalsy();
    expect(newState).toMatchSnapshot();
  });

  test('cardSettings/TOGGLE_SHOWN_DISABLE_CARD_LIMITS', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_DISABLE_CARD_LIMITS,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownDisableLimits).toBeFalsy();
  });

  test('cardSettings/TOGGLE_SHOWN_DISABLE_CARD_LIMITS_CONFIRM', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_DISABLE_CARD_LIMITS_CONFIRM,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownDisableLimitsConfirm).toBeFalsy();
  });

  test('cardSettings/TOGGLE_SHOWN_INCREASE_CARD_LIMIT', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_INCREASE_CARD_LIMIT,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownIncreaseLimit).toBeFalsy();
  });

  test('cardSettings/TOGGLE_SHOWN_INCREASE_CARD_LIMIT_CONFIRM', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_INCREASE_CARD_LIMIT_CONFIRM,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.modals.isShownIncreaseLimitConfirm).toBeFalsy();
  });

  test('cardSettings/TOGGLE_SHOWN_CARD_LIMITS_TIMER', () => {
    const action: IToggleCardModal = {
      type: TOGGLE_SHOWN_CARD_LIMITS_TIMER,
      payload: false,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState.limitsSettings.isShownDisableLimitsTimer).toBeFalsy();
  });

  test('cardSettings/REDUCTION_CARD_LIMIT', () => {
    const action: IIncreaseCardLimitAction = {
      type: `${REDUCTION_CARD_LIMIT}_SUCCESS`,
      payload: { data: cardData },
      meta: accountId,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState).toMatchSnapshot();
  });

  test('cardSettings/SET_INCREASED_CARD_LIMIT', () => {
    const action: ISetIncreasedLimitsAction = {
      type: `${SET_INCREASED_CARD_LIMIT}_SUCCESS`,
      payload: cardLimit,
    };

    const newState = cardSettings(initCardState, action);
    expect(newState).toMatchSnapshot();
  });
});
