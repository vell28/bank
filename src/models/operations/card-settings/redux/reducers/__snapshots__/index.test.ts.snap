// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardSettings reducers cardSettings/DISABLE_CARD_LIMITS_CONFIRM 1`] = `
Object {
  "data": Object {},
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": false,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": true,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": true,
  },
  "selectedCard": Object {
    "account123": Object {
      "card": Object {
        "accountId": "PPY5107",
        "availableBalance": 1897.61,
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "clientId": "6449",
        "currency": Object {
          "code": "EUR",
        },
        "expireAt": "2022-05-01T00:00:00",
        "id": "PPY13643",
        "limits": Array [
          Object {
            "maxValue": 500,
            "title": null,
            "type": "DAY",
            "usedValue": 0,
          },
        ],
        "number": "5277 60_3 984",
        "options": Object {
          "limits": Object {
            "disable": false,
          },
        },
        "owner": Object {
          "embossedName": "RALF WILLIAMS",
        },
        "paySystem": "MASTERCARD",
        "plasticStatus": "PENDING_SHIPMENT",
        "productType": "MAIN",
        "status": "ACTIVE",
      },
    },
  },
}
`;

exports[`cardSettings reducers cardSettings/FETCH_CARD_UNMASKED 1`] = `
Object {
  "data": Object {
    "test123": Object {
      "cvc": "675",
      "expireAt": "2022-05-01T00:00:00",
      "number": "**** **10 0004 ****",
      "owner": Object {
        "embossedName": "RALF WILLIAMS",
      },
      "pin": null,
    },
  },
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": true,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": true,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": true,
  },
  "selectedCard": Object {},
}
`;

exports[`cardSettings reducers cardSettings/INCREASE_CARD_LIMIT_CONFIRM 1`] = `
Object {
  "data": Object {},
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": true,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": false,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": true,
  },
  "selectedCard": Object {
    "account123": Object {
      "card": Object {
        "accountId": "PPY5107",
        "availableBalance": 1897.61,
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "clientId": "6449",
        "currency": Object {
          "code": "EUR",
        },
        "expireAt": "2022-05-01T00:00:00",
        "id": "PPY13643",
        "limits": Array [
          Object {
            "maxValue": 500,
            "title": null,
            "type": "DAY",
            "usedValue": 0,
          },
        ],
        "number": "5277 60_3 984",
        "options": Object {
          "limits": Object {
            "disable": false,
          },
        },
        "owner": Object {
          "embossedName": "RALF WILLIAMS",
        },
        "paySystem": "MASTERCARD",
        "plasticStatus": "PENDING_SHIPMENT",
        "productType": "MAIN",
        "status": "ACTIVE",
      },
    },
  },
}
`;

exports[`cardSettings reducers cardSettings/LOCK_CARD 1`] = `
Object {
  "data": Object {},
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": true,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": true,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": true,
  },
  "selectedCard": Object {
    "account123": Object {
      "card": Object {
        "accountId": "PPY5107",
        "availableBalance": 1897.61,
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "clientId": "6449",
        "currency": Object {
          "code": "EUR",
        },
        "expireAt": "2022-05-01T00:00:00",
        "id": "PPY13643",
        "limits": Array [
          Object {
            "maxValue": 500,
            "title": null,
            "type": "DAY",
            "usedValue": 0,
          },
        ],
        "number": "5277 60_3 984",
        "options": Object {
          "limits": Object {
            "disable": false,
          },
        },
        "owner": Object {
          "embossedName": "RALF WILLIAMS",
        },
        "paySystem": "MASTERCARD",
        "plasticStatus": "PENDING_SHIPMENT",
        "productType": "MAIN",
        "status": undefined,
      },
    },
  },
}
`;

exports[`cardSettings reducers cardSettings/REDUCTION_CARD_LIMIT 1`] = `
Object {
  "data": Object {},
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": true,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": true,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": true,
  },
  "selectedCard": Object {
    "account123": Object {
      "card": Object {
        "accountId": "PPY5107",
        "availableBalance": 1897.61,
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "clientId": "6449",
        "currency": Object {
          "code": "EUR",
        },
        "expireAt": "2022-05-01T00:00:00",
        "id": "PPY13643",
        "limits": Array [
          Object {
            "maxValue": 500,
            "title": null,
            "type": "DAY",
            "usedValue": 0,
          },
        ],
        "number": "5277 60_3 984",
        "options": Object {
          "limits": Object {
            "disable": false,
          },
        },
        "owner": Object {
          "embossedName": "RALF WILLIAMS",
        },
        "paySystem": "MASTERCARD",
        "plasticStatus": "PENDING_SHIPMENT",
        "productType": "MAIN",
        "status": "ACTIVE",
      },
    },
  },
}
`;

exports[`cardSettings reducers cardSettings/SET_INCREASED_CARD_LIMIT 1`] = `
Object {
  "data": Object {},
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": false,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": true,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": true,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": true,
  },
  "selectedCard": Object {},
}
`;

exports[`cardSettings reducers cardSettings/UNLOCK_CARD_CONFIRM 1`] = `
Object {
  "data": Object {},
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
  "limitsSettings": Object {
    "increasedLimit": Object {
      "maxValue": "",
      "type": "DAY",
    },
    "isShownDisableLimitsTimer": true,
  },
  "modals": Object {
    "isShownDisableLimits": true,
    "isShownDisableLimitsConfirm": true,
    "isShownIncreaseLimit": true,
    "isShownIncreaseLimitConfirm": true,
    "isShownLock": true,
    "isShownUnLock": true,
    "isShownUnLockConfirm": false,
  },
  "selectedCard": Object {
    "account123": Object {
      "card": Object {
        "accountId": "PPY5107",
        "availableBalance": 1897.61,
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "clientId": "6449",
        "currency": Object {
          "code": "EUR",
        },
        "expireAt": "2022-05-01T00:00:00",
        "id": "PPY13643",
        "limits": Array [
          Object {
            "maxValue": 500,
            "title": null,
            "type": "DAY",
            "usedValue": 0,
          },
        ],
        "number": "5277 60_3 984",
        "options": Object {
          "limits": Object {
            "disable": false,
          },
        },
        "owner": Object {
          "embossedName": "RALF WILLIAMS",
        },
        "paySystem": "MASTERCARD",
        "plasticStatus": "PENDING_SHIPMENT",
        "productType": "MAIN",
        "status": "ACTIVE",
      },
    },
  },
}
`;
